### ISSUE TITLE
More complete example needed for symfony/routing

### ISSUE THREAD

**Issue by udf2457** [symfony/symfony-docs#15076]
The documentation is a little bit lacking in terms of using `symfony/routing` as a standalone component rather than as part of a `symfony` install.

Specifically I find it deeply unclear as to what exactly the recommended code pattern would be for the most obvious use-case, namely implementation in an `index.php` file as the basis of a site route.

The code example on the github repo (https://github.com/symfony/routing) and at the top of the linked-to docs (https://symfony.com/doc/current/routing.html) seems to me to be barely half implemented.  It mentions `RequestContext()` ,`UrlMatcher()` and `UrlGenerator()` but then stops abruptly short of actually giving a cohesive example of how you would tie these together in a basic functional use-case such as an `index.php` implementation.

Searches on the internet bring up answers related to previous versions which indicate further use of `match()`, but then the docs page makes no mention of `match()` so the reader is left wondering if `match()` is still used and how its used ?

I hope you can improve the docs.

**Comment by David-Maran** [comment]
Because this documentation is not how to dev in php or install and config dependencies. 
You can discover some symfony basics here : https://symfony.com/doc/current/introduction/from_flat_php_to_symfony.html

**Comment by udf2457** [comment]
If you are saying `symfony/routing` can be used standalone then surely it is fair and reasonable to expect a workable basic example for the most common use-case (i.e. index.php) to be provided in the docs ?  Giving a half-hearted example in the docs helps nobody, you might as well not bother publishing any example code at all.

**Comment by David-Baron** [comment]
Complete list here: https://symfony.com/components/Routing

**Comment by David-Maran** [comment]
Complete doc to use is here: https://symfony.com/doc/current/routing.html#creating-routes-in-yaml-xml-or-php-files and yes examples inside.
Maybe here: https://stackoverflow.com/search?q=symfony+routing you can find answers to your questions. Otherwise ask yours.

**Comment by David-Maran** [comment]
And because I decided to be nice today, here is an example to use like the getting started say here https://github.com/symfony/routing :
```
$routes = new RouteCollection();
$routes->add('blog_add', new Route('/blog/add', [
    'controller' => 'BlogController::add',
]));

// index.php

$request = Request::createFromGlobals();
$context = new RequestContext();
$context->fromRequest($request);
$matcher = new UrlMatcher($routes, $context);

try {
    $attributes = $matcher->match($request->getPathInfo());

    $response = $attributes['controller']($request);
} catch (ResourceNotFoundException $exception) {
    $response = new Response('Not Found', 404);
} catch (Exception $exception) {
    $response = new Response('An error occurred', 500);
}

$response->send();
```

**Comment by udf2457** [comment]
FYI I did search stackoverflow but the vast majority of content relates to long obsolete versions of routing (e.g. v2).

Thank you for the example.  Despite having spent many hours previously reading the docs (including the page you refer to), I would never have come to the conclusion of that pattern being the one to follow (and I'm not exactly a PHP noob).

**Comment by David-Maran** [comment]
If it's ok for you, closed the issue please

**Comment by WaqasBukhary** [comment]
I checked and unable to make it work as standalone component
Request class in the above example by @David-Baron is not part of symfony/routing so that example is wrong. It will need all of symfony

I also tried using this example in this link. It just go over in all the lines in xdebug and does not call BlogController
https://github.com/symfony/routing

There should be some basic working example to make it work from index.php file if only symfony/routing is installed.

**Comment by David-Maran** [comment]
Hmm... read more the document and follow the links I give please.
You can use most of symfony modules in a non symfony project, for that you need to understand php and follow the symfony documentation.
And my example works very well, of course it is to be adapted according to the version of symfony you are using and to the structure of your project.

This discussion is from 2021 and closed, if you have any difficulties please ask your question on stackoverflow or create an issue if you think the symfony code has an error.

Greetings.

**Comment by WaqasBukhary** [comment]
Maybe you are right. I invest few hours on the links and gpt but was not able to set it up. Getting frustrated I made the comments. The more I was reading the more it's dependent on other things. For e.g. symfony docs explain bin/console debug:router will list all routes. But there was no bin/console dir with symfony router. If it's dependent on other symfony packages then composer file should include them so related packages such as request and console are installed with them.

I quickly switched to laminas-router that day and it worked for me. When I ran laminas-router command the composer installed the following laminas packages laminas-escaper, http, loader, servicemanager, stdlib, uri, and laminas-validator.

Thanks for the help anyways.
